tidb-cloud: update Vercel integration connecting options (#13872)
pingcap/docs: tidb-cloud/integrate-tidbcloud-with-vercel.md

* [Connect via the TiDB Cloud Vercel integration](#connect-via-the-tidb-cloud-vercel-integration)
* [Connect via manually configuring environment variables](#connect-via-manually-setting-environment-variables)

## Choosing a connection method

> TiDB Cloud provides several options for programmatically connecting to your database:

Reviewer: Choosing a connection method 看起来和前面的有点混淆
![image](https://github.com/pingcap/docs/assets/79440533/da736df8-aa10-47df-81a5-5c3ee05904ea)

Reviewer: ```suggestion
For both of the preceding methods, TiDB Cloud provides the following options for programmatically connecting to your database:
```

> - Direct connection: Connect your TiDB Cloud cluster to your Vercel project directly using MySQL's standard connection system.
> - [Data App](/tidb-cloud/data-service-manage-data-app.md): Access data of TiDB Cloud cluster through a group of HTTP endpoints.

Reviewer: ```suggestion
- Direct connection: connect your TiDB Cloud cluster to your Vercel project directly using MySQL's standard connection system.
- [Data App](/tidb-cloud/data-service-manage-data-app.md): access data of your TiDB Cloud cluster through a group of HTTP endpoints.
```

1. Click **Add Integration** in the upper-right area of the [TiDB Cloud Vercel integration](https://vercel.com/integrations/tidb-cloud) page. The **Add TiDB Cloud** dialog is displayed.
2. Select the scope of your integration in the drop-down list and click **Continue**.
3. Select the Vercel projects to which the integration will be added and click **Continue**.
4. Confirm the required permissions for integration and click **Add Integration**. Then you are directed to an integration page of the TiDB Cloud console.
5. On the integration page, do the following:

    1. Select your target Vercel projects and click **Next**.
    2. Select your target TiDB Cloud organization and project.
    3. Select **Cluster** as your connection type.
    4. Select your target TiDB Cloud cluster. If the **Cluster** drop-down list is empty or you want to select a new TiDB Serverless cluster, click **+ Create Cluster** in the list to create one.
    5. Select the framework that your Vercel projects are using. If the target framework is not listed, select **General**. Different frameworks determine different environment variables.
    6. Click **Add Integration and Return to Vercel**.

> ![Vercel Integration Page](/media/tidb-cloud/vercel/integration-link-cluster-page.png)

Reviewer 2: This image should also be indented by four spaces, but I'm not sure whether it can display well now, considering a recent error. @qiancai What do you think?

- [Create a TiDB Serverless cluster](/tidb-cloud/create-tidb-cluster-serverless.md)
- [Create a TiDB Dedicated cluster](/tidb-cloud/create-tidb-cluster.md)

> **Note:**
>

> > Currently, integration supports create TiDB serverless cluster. You can create during the integration process.

Reviewer: ```suggestion
> The TiDB Cloud Vercel integration supports creating TiDB Serverless clusters. You can create one during the integration process.
```

To [integrate with Vercel via the TiDB Cloud Vercel Integration](#connect-via-the-tidb-cloud-vercel-integration), you are expected to have the "Owner" access to your organization or the "Member" access to the target project in TiDB Cloud. For more information, see [Manage role access](/tidb-cloud/manage-user-access.md#manage-role-access).

One TiDB Cloud cluster can connect to multiple Vercel projects.

<SimpleTab>
<div label="Direct connection">

### All IP addresses allowed for traffic filter in TiDB Cloud

For TiDB Dedicated clusters, make sure that the traffic filter of the cluster allows all IP addresses (set to `0.0.0.0/0`) for connection, this is because Vercel deployments use [dynamic IP addresses](https://vercel.com/guides/how-to-allowlist-deployment-ip-address). If you use the TiDB Cloud Vercel integration, TiDB Cloud automatically adds a `0.0.0.0/0` traffic filter to your cluster in the integration workflow if there is none.

TiDB Serverless clusters allow all IP addresses for connection by default, so you do not need to configure any traffic filter.

</div>

<div label="Data App">

> ### A Data App and endpoints

Reviewer: 目前文档站的 simple tab 下不支持加标题，会造成右侧本页导航同时显示两个标题，但是点击不是当前 tab 的那个标题无效
![image](https://github.com/pingcap/docs/assets/79440533/4fc181f1-89e8-4d19-be17-bfe3ab4a6451)

1. [Create a Data App](/tidb-cloud/data-service-manage-data-app.md#create-a-data-app) on the [**Data Service**](https://tidbcloud.com/console/data-service) page of your project.
2. [Link the Data App](/tidb-cloud/data-service-manage-data-app.md#manage-linked-data-sources) to the target TiDB Cloud cluster.

> 3. [Manage endpoints](/tidb-cloud/data-service-manage-endpoint.md) that you can customize to execute SQL statements.

Reviewer: ```suggestion
3. [Manage endpoints](/tidb-cloud/data-service-manage-endpoint.md) so that you can customize them to execute SQL statements.
```

> One Vercel project can only connect to one TiDB Cloud Data App. To change the integration, you need to first disconnect the current app and then connect to a new app.

Reviewer: ```suggestion
One Vercel project can only connect to one TiDB Cloud Data App. To change the Data App for your Vercel project, you need to first disconnect the current App and then connect to a new App.
```

</div>
</SimpleTab>

## Connect via the TiDB Cloud Vercel integration

To connect via the TiDB Cloud Vercel integration, go to the [TiDB Cloud integration](https://vercel.com/integrations/tidb-cloud) page from the [Vercel's Integrations Marketplace](https://vercel.com/integrations). Using this method, you can choose which cluster to connect to, and TiDB Cloud will automatically generate all the necessary environment variables for your Vercel projects.

The detailed steps are as follows:

<SimpleTab>
<div label="Direct connection">

1. Click **Add Integration** in the upper-right area of the [TiDB Cloud Vercel integration](https://vercel.com/integrations/tidb-cloud) page. The **Add TiDB Cloud** dialog is displayed.
2. Select the scope of your integration in the drop-down list and click **Continue**.
3. Select the Vercel projects to which the integration will be added and click **Continue**.
4. Confirm the required permissions for integration and click **Add Integration**. Then you are directed to an integration page of the TiDB Cloud console.

> 5. Select the target Vercel projects and click **Next**. 
> 6. Select the target TiDB Cloud Organization and TiDB Cloud Project.
> 7. Select the Connection Type **Cluster**. 
> 8. Select the target TiDB Cloud cluster. If you do not have any, drop down and click **+ Create Cluster** to create one.
> 9. Select the framework your Vercel projects using. If the framework isn't listed, select **General**. Different frameworks determine different environment variables.
> 10. Click **Add Integration and Return to Vercel**.
> ![Vercel Integration Page](/media/tidb-cloud/vercel/integration-link-cluster-page.png)

Reviewer: ```suggestion
5. On the integration page, do the following:

    1. Select your target Vercel projects and click **Next**. 
    2. Select your target TiDB Cloud organization and project.
    3. Select **Cluster** as your connection type.
    4. Select your target TiDB Cloud cluster. If the **Cluster** drop-down list is empty or you want to select a new TiDB Serverless cluster, click **+ Create Cluster** in the list to create one.
    5. Select the framework that your Vercel projects are using. If the target framework is not listed, select **General**. Different frameworks determine different environment variables.
    6. Click **Add Integration and Return to Vercel**.

![Vercel Integration Page](/media/tidb-cloud/vercel/integration-link-cluster-page.png)

```

> 11. Back to your Vercel dashboard, go to your Vercel project, click **Settings** > **Environment Variables**, and confirm that the environment variables have been automatically added.

Reviewer: ```suggestion
11. Get back to your Vercel dashboard, go to your Vercel project, click **Settings** > **Environment Variables**, and confirm that the environment variables for your target TiDB cluster have been automatically added.
```

2. Select the scope of your integration in the drop-down list and click **CONTINUE**.
3. Select the Vercel Projects to which the integration will be added and click **CONTINUE**.
4. Confirm the required permissions for integration and click **ADD INTEGRATION**. Then you are directed to an integration page of the TiDB Cloud console.
5. Select the target Vercel projects and click **Next**. 
6. Select the target TiDB Cloud Organization and TiDB Cloud Project.
7. Select the Connection Type **Cluster**. 
8. Select the target TiDB Cloud cluster.
9. Select the framework your Vercel projects using. If the framework isn't listed, select **General**. Different frameworks determine different environment variables.
10. Click **Add Integration and Return to Vercel**.
![Vercel Integration Page](/media/tidb-cloud/vercel/integration-link-cluster-page.png)
11. Back to your Vercel dashboard, go to your Vercel project, click **Settings** > **Environment Variables**, and confirm that the environment variables have been automatically added.

>     If the variables have been added, the connection is completed.

Reviewer 3: ```suggestion
    If the variables have been added, the integration is completed.
```

DATABASE_URL
```

</div>

<div label="Data App">

1. Click **Add Integration** in the upper-right area of the [TiDB Cloud Vercel integration](https://vercel.com/integrations/tidb-cloud) page. The **Add TiDB Cloud** dialog is displayed.
2. Select the scope of your integration in the drop-down list and click **Continue**.
3. Select the Vercel projects to which the integration will be added and click **Continue**.
4. Confirm the required permissions for integration and click **Add Integration**. Then you are directed to an integration page of the TiDB Cloud console.

> 5. Select the target Vercel projects and click **Next**.
> 6. Select the target TiDB Cloud Organization and TiDB Cloud Project.
> 7. Select the Connection Type **Data App**.
> 8. Select the target TiDB Cloud Data App. 
> 9. Click **Add Integration and Return to Vercel**.
> ![Vercel Integration Page](/media/tidb-cloud/vercel/integration-link-data-app-page.png)

Reviewer: ```suggestion
5. On the integration page, do the following:

    1. Select your target Vercel projects and click **Next**. 
    2. Select your target TiDB Cloud organization and project.
    3. Select **Data App** as your connection type.
    4. Select your target TiDB Data App. 
    6. Click **Add Integration and Return to Vercel**.

![Vercel Integration Page](/media/tidb-cloud/vercel/integration-link-data-app-page.png)

```

> 10. Back to your Vercel dashboard, go to your Vercel project, click **Settings** > **Environment Variables**, and confirm that the environment variables have been automatically added.

Reviewer: ```suggestion
10. Get back to your Vercel dashboard, go to your Vercel project, click **Settings** > **Environment Variables**, and check whether the environment variables for your target Data App have been automatically added.
```

>    If the variables have been added, the connection is completed.
> 
> After you have completed the integration setup and successfully connected a TiDB Cloud Data App to your Vercel projects, the information necessary for the connection is automatically set in the projects' environment variables.

Reviewer: ```suggestion
   If the following variables have been added, the integration is completed.
```
